File: sdk-objectivec/office365_odata_base/office365_odata_impl/Implementations/MSRequestImpl.m
/*******************************************************************************
 * Copyright (c) Microsoft Open Technologies, Inc.
 * All Rights Reserved
 * See License.txt in the project root for license information.
 ******************************************************************************/

#import "MSRequestImpl.h"

@interface MSRequestImpl()

@property NSMutableURLRequest* request;
@property MSHttpVerb httpVerb;

@end

@implementation MSRequestImpl

-(id)init;{
    self.request = [[NSMutableURLRequest alloc]init];
    
    return self;
}

-(void) setContent : (NSData*) content{
    self.request.HTTPBody = content;
}

-(NSData*) getContent{
    return self.request.HTTPBody;
}

-(NSDictionary *) getHeaders{
    return [self.request allHTTPHeaderFields];
}

-(void)setHeaders :(NSArray*) headers{
    for (NSDictionary* dicc in headers) {
        [self.request addValue:[[dicc allKeys] objectAtIndex:0] forHTTPHeaderField:[[dicc allValues] objectAtIndex:0]];
    }
}

-(void)addHeader : (NSString*) name : (NSString*) value{
    [self.request addValue:value forHTTPHeaderField: name];
}

-(void)removeHeader : (NSString*) name{
}

-(MSHttpVerb) getVerb{
    return self.httpVerb;
}

-(void)setVerb : (MSHttpVerb) httpVerb{
    self.httpVerb = httpVerb;
    self.request.HTTPMethod = [self verbToString :httpVerb];
}

-(void)setUrl : (NSString*) url{
    [self.request setURL:[[NSURL alloc] initWithString:url]];
}

-(NSMutableURLRequest*)getMutableRequest{
    return self.request;
}

-(NSString *)getUrl{
    return self.request.URL.absoluteString;
}

-(NSString*)verbToString : (MSHttpVerb) verb{

    NSString* verbToString;
    
    switch (verb) {
        case GET:
            verbToString = @"GET";
            break;
        case POST:
            verbToString = @"POST";
            break;
        case DELETE:
            verbToString = @"DELETE";
            break;
        case PATCH:
            verbToString = @"PATCH";
            break;
        case PUT:
            verbToString = @"PUT";
            break;
        case HEAD:
            verbToString = @"HEAD";
            break;
        case OPTIONS:
            verbToString = @"OPTIONS";
            break;
        default:
            break;
    }
    
    return verbToString;
}

@end

File: sdk-swift/office365_odata_impl/MSRequestImpl.swift
/*******************************************************************************
* Copyright (c) Microsoft Open Technologies, Inc.
* All Rights Reserved
* Licensed under the Apache License, Version 2.0.
* See License.txt in the project root for license information.
******************************************************************************/

import Foundation

public class MSRequestImpl : MSRequest{

    private var request =  NSMutableURLRequest();
    private var httpVerb : MSHttpVerb?;
    
    required public init(){
    
    }
    
    public func setContent(content: NSData?) {
        
        self.request.HTTPBody = content;
    }
  
    public func getContent() -> NSData {
        
        return self.request.HTTPBody!;
    }
    
    public func getHeaders() -> NSDictionary {
        
        return self.request.allHTTPHeaderFields!;
    }
    
    public func setHeaders(headers: NSArray) {

        for dicc in headers{
            var diccionary = dicc as NSDictionary;
            self.request.addValue(diccionary.allKeys[0] as? String, forHTTPHeaderField: diccionary.allValues[0] as String);
        }
    }
    
    public func addHeader(name: String, value: String) {
        self.request.addValue(value, forHTTPHeaderField: name);
    }
    
    public func removeHeader(name: String) {
        
    }
    
    public func getVerb() -> MSHttpVerb {
        
        return self.httpVerb!;
    }
    
    public func setVerb(httpVerb: MSHttpVerb) {
     
        self.request.HTTPMethod = httpVerb.rawValue;
    }
    
    public func setUrl(url: String) {
    
        self.request.URL = NSURL(string: url);
    }
    
    
    public func getMutableRequest() -> NSMutableURLRequest{
        
        return self.request;
    }
    
    public func getUrl() -> NSString {
        
        return self.request.URL!.absoluteString!;
    }
}